### DISCUSSION TITLE
socket.accept() out of system resource

### DISCUSSION THREAD

**Discussion by BojingLiCCNU** [fastapi/fastapi#8692] [Questions]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
@app.post("/coagulantPrediction")
async def create_item(item: Item):
    starttime = time.strftime("%a %b %d %H:%M:%S %Y", time.localtime())
    print('starttime', starttime)
    rawjon = item.dict()
    all_pool_data = rawjon['dataInput']
    all_pool_result = []
    for i in range(0, len(all_pool_data)):
        one_pool_data = all_pool_data[i]
        one_pool_result = doPrediction(one_pool_data, MODEL_COA, MODEL_TUR)
        one_pool_result_dict = unparse_result(one_pool_result, one_pool_data)
        all_pool_result.append(one_pool_result_dict)
    rawjon.update({"dataOutput": all_pool_result})
    gc.collect()
    return rawjon
```

### Description

I started post api and tested successed in postman. I write a timed program to call it. 
But after 24 to 48 hours, the error happened 

"
socket.accept() out of system resource
OSError: [Errno 24] Too many open files
"
I checked that all the IO stream is closed and I don't know  how to fix it.
The solution on google tells me to modify the number of ulimit in linux system.
But this way just append the error from 48 hours to 72 hours.
Could you help me ,thank you .

### Operating System

Linux

### Operating System Details

LINUX

### FastAPI Version

0.70.0

### Python Version

3.8

### Additional Context

empty

**Comment by Dustyposa** [comment]
Maybe you need check the file IO status again.
I can't get more info from you example code. : (

**Comment by kokman** [comment]
I also encountered the same problem, how to solve it?

2024-05-17 11:13:45,915 - ERROR - base_events - default_exception_handler - line:1758 - socket.accept() out of system resource
socket: <asyncio.TransportSocket fd=8, family=AddressFamily.AF_INET, type=SocketKind.SOCK_STREAM, proto=6, laddr=('0.0.0.0', 8100)>
Traceback (most recent call last):
  File "asyncio/selector_events.py", line 159, in _accept_connection
  File "socket.py", line 293, in accept
OSError: [Errno 24] Too many open files

**Comment by YuriiMotov** [comment]
It's likely a problem in user code, and it's impossible to reproduce it with given code example
